full-service/src/service/wallet_impl.rs

```diff
@@ -198,41 +207,42 @@ impl<
                             &AccountID(a.account_id_hex.clone()),
                             local_height,
                             network_height,
-                            &conn,
+                            &conn_manager,
                         )
                         .map_err(|e| e.into())
                     })
                     .collect::<Result<Vec<JsonAccount>, WalletServiceError>>()
-            })?,
-        )
+            })?)
     }
 
     pub fn update_account_name(
         &self,
         account_id_hex: &str,
         name: String,
     ) -> Result<JsonAccount, WalletServiceError> {
-        let conn = self.wallet_db.get_conn()?;
+        let conn_manager = self.wallet_db.get_conn_manager()?;
 
-        Ok(conn.transaction::<JsonAccount, WalletServiceError, _>(|| {
-            Account::get(&AccountID(account_id_hex.to_string()), &conn)?
-                .update_name(name, &conn)?;
-
-            let local_height = self.ledger_db.num_blocks()?;
-            let network_state = self.network_state.read().expect("lock poisoned");
-            // network_height = network_block_index + 1
-            let network_height = network_state
-                .highest_block_index_on_network()
-                .map(|v| v + 1)
-                .unwrap_or(0);
-            let decorated_account = Account::get_decorated(
-                &AccountID(account_id_hex.to_string()),
-                local_height,
-                network_height,
-                &conn,
-            )?;
-            Ok(decorated_account)
-        })?)
+        Ok(conn_manager
+            .conn
+            .transaction::<JsonAccount, WalletServiceError, _>(|| {
+                Account::get(&AccountID(account_id_hex.to_string()), &conn_manager.conn)?
+                    .update_name(name, &conn_manager.conn)?;
+
+                let local_height = self.ledger_db.num_blocks()?;
```

Comment:
Its pretty weird to have all this stuff inside a method called `update_account_name`. Why is this here? Is this duplicated from anywhere?

Reply:
Maybe there should be a `self.get_decorated_account(account_id)` if you want to return the entire account as a response.